Mathlib/CategoryTheory/Sites/IsSheafOneHypercover.lean

```diff
@@ -0,0 +1,167 @@
+  le {X : C} (S : Sieve X) (hS : S ∈ J X) :
+    ∃ (E : J.OneHypercover X) (_ : H E), E.sieve₀ ≤ S
+
+variable [H.IsGenerating]
+
+lemma exists_oneHypercover {X : C} (S : Sieve X) (hS : S ∈ J X) :
+    ∃ (E : J.OneHypercover X) (_ : H E), E.sieve₀ ≤ S :=
+  IsGenerating.le _ hS
+
+variable (P : Cᵒᵖ ⥤ A)
+
+namespace IsSheafIff
+
+variable (hP : ∀ ⦃X : C⦄ (E : J.OneHypercover X) (_ : H E), Nonempty (IsLimit (E.multifork P)))
+
+lemma hom_ext {X : C} (S : Sieve X) (hS : S ∈ J X) {T : A}
+    {x y : T ⟶ P.obj (Opposite.op X)}
+    (h : ∀ ⦃Y : C⦄ (f : Y ⟶ X) (_ : S f), x ≫ P.map f.op = y ≫ P.map f.op) :
+    x = y := by
+  obtain ⟨E, hE, le⟩ := H.exists_oneHypercover S hS
+  exact Multifork.IsLimit.hom_ext (hP E hE).some (fun j => h _ (le _ (Sieve.ofArrows_mk _ _ _)))
+
+variable {P H}
+variable {X : C} {S : Sieve X} {E : J.OneHypercover X} (hE : H E) (le : E.sieve₀ ≤ S)
+
+section
+
+variable (F : Multifork (Cover.index ⟨S, J.superset_covering le E.mem₀⟩ P))
+
+/-- Auxiliary definition for `isLimit`. -/
+noncomputable def lift : F.pt ⟶ P.obj (Opposite.op X) :=
+  Multifork.IsLimit.lift (hP E hE).some
+    (fun i => F.ι ⟨_, E.f i, le _ (Sieve.ofArrows_mk _ _ _)⟩)
+    (fun ⟨⟨i₁, i₂⟩, j⟩ => F.condition
+        { h₁ := le _ ((Sieve.ofArrows_mk _ _ i₁))
+          h₂ := le _ ((Sieve.ofArrows_mk _ _ i₂))
+          w := E.w j })
+
+@[reassoc]
+lemma fac' (i : E.I₀) :
+    lift hP hE le F ≫ P.map (E.f i).op =
+      F.ι ⟨_, E.f i, le _ (Sieve.ofArrows_mk _ _ _)⟩ :=
+  Multifork.IsLimit.fac (hP E hE).some _ _ i
+
+lemma fac {Y : C} (f : Y ⟶ X) (hf : S f) :
+    lift hP hE le F ≫ P.map f.op = F.ι ⟨Y, f, hf⟩ := by
+  apply hom_ext H P hP _ (J.pullback_stable f E.mem₀)
+  intro Z g
+  rintro ⟨T, a, b, hb, fac⟩
+  obtain ⟨i, rfl, hi⟩ := hb.exists
+  dsimp at hi
+  rw [id_comp] at hi
+  subst hi
+  rw [assoc, ← P.map_comp, ← op_comp, ← fac,
+    op_comp, P.map_comp, fac'_assoc]
+  exact F.condition
+    { h₁ := le _ (Sieve.ofArrows_mk _ _ _)
+      h₂ := hf
+      w := fac }
+
+end
+
+/-- Auxiliary definition for `OneHypercoverFamily.isSheaf_iff`. -/
+noncomputable def isLimit :
+    IsLimit (Cover.multifork ⟨S, J.superset_covering le E.mem₀⟩ P) :=
+  Multifork.IsLimit.mk _
+    (fun F => lift hP hE le F)
+    (fun F => by
+      rintro ⟨Y, f, hf⟩
+      apply fac)
+    (fun F m hm => by
+      apply hom_ext H P hP S (J.superset_covering le E.mem₀)
+      intro Y f hf
+      rw [fac _ _ _ _ _ hf]
+      exact hm ⟨_, _, hf⟩)
+
+end IsSheafIff
+
+lemma isSheaf_iff :
+    Presheaf.IsSheaf J P ↔ ∀ ⦃X : C⦄ (E : J.OneHypercover X)
+      (_ : H E), Nonempty (IsLimit (E.multifork P)) := by
+  constructor
+  · intro hP X E _
+    exact ⟨E.isLimitMultifork ⟨_, hP⟩⟩
+  · intro hP
+    rw [Presheaf.isSheaf_iff_multifork]
+    rintro X ⟨S, hS⟩
+    obtain ⟨E, hE, le⟩ := H.exists_oneHypercover S hS
+    exact ⟨IsSheafIff.isLimit hP hE le⟩
+
+end OneHypercoverFamily
+
+/-- Condition that a topology is generated by 1-hypercovers of size `w`. -/
```

Comment:
```suggestion
/-- Condition that a topology is generated by the 1-hypercovers of size `w`. -/
```

Reply:
Doesn't it mean that it's generated by *all* the 1-hypercovers?

Reply:
Yes, `IsGeneratedByOneHypercovers` means it is generated by *all the* 1-hypercovers.

Reply:
This is why I suggest adding "the". But I'm not a native English speaker, and in any case it's not that important.